noxfiles/ci_nox.py

```diff
@@ -0,0 +1,163 @@
+"""Contains the nox sessions used during CI checks."""
+import nox
+from constants_nox import CI_ARGS, COMPOSE_SERVICE_NAME, RUN_NO_DEPS, START_APP
+from docker_nox import build
+from run_infrastructure import run_infrastructure
+
+
+@nox.session()
+def ci_suite(session: nox.Session) -> None:
+    """
+    Runs all of the CI checks, except for 'pytest_external'.
+
+    Excludes external tests so that no additional secrets/tooling are required.
+    """
+    # Use "notify" instead of direct calls here to provide better user feedback
+    session.notify("teardown")
+    session.notify("build", ["test"])
+    session.notify("black")
+    session.notify("isort")
+    session.notify("xenon")
+    session.notify("mypy")
+    session.notify("pylint")
+    session.notify("check_install")
+    session.notify("pytest_unit")
+    session.notify("pytest_integration")
+    session.notify("teardown")
+
+
+# Static Checks
+@nox.session()
+def black(session: nox.Session) -> None:
+    """Run the 'black' style linter."""
+    command = (
+        *RUN_NO_DEPS,
+        "black",
+        "--check",
+        "src",
+        "tests",
+        "noxfiles",
+    )
+    session.run(*command, external=True)
+
+
+@nox.session()
+def isort(session: nox.Session) -> None:
+    """Run the 'isort' import linter."""
+    command = (*RUN_NO_DEPS, "isort", "src", "tests", "noxfiles", "--check")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def mypy(session: nox.Session) -> None:
+    """Run the 'mypy' static type checker."""
+    command = (*RUN_NO_DEPS, "mypy")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def pylint(session: nox.Session) -> None:
+    """Run the 'pylint' code linter."""
+    command = (*RUN_NO_DEPS, "pylint", "src", "noxfiles")
+    session.run(*command, external=True)
+
+
+@nox.session()
+def xenon(session: nox.Session) -> None:
+    """Run 'xenon' code complexity monitoring."""
+    command = (
+        "xenon",
+        "noxfiles",
+        "src",
+        "tests",
+        "--max-absolute B",
+        "--max-modules B",
+        "--max-average A",
+        "--ignore 'data, docs'",
+        "--exclude src/fidesops/_version.py",
+    )
+    session.run(*command, external=True)
```

Comment:
oh right

Are you ok if I add this to the `dev-requirements` then since when I ran it everything passed and therefore no code changes would be required? or I can take it out and we can add it in another PR

Reply:
added it to the dev-requirements and updated the command to run in docker, succeeds now

Reply:
I have no issues with this.